xla/service/gpu/fusions/dynamic_slice_fusion_test.cc

```diff
@@ -2996,6 +2996,417 @@ TEST_F(DynamicSliceFusionTest, ReduceScatterDUSLoopIterationOffset) {
+      param = (s32[], s32[16, 4], s32[8, 4], s32[]) parameter(0)
+      i = s32[] get-tuple-element(param), index=0
+      dest = s32[16,4] get-tuple-element(param), index=1
+      src = s32[8,4] get-tuple-element(param), index=2
+      loop_iter = s32[] get-tuple-element(param), index=3
+      zero = u64[] constant(0)
+      fusion = s32[16,4] fusion(dest, src, loop_iter, zero), kind=kCustom, calls=dynamic-slice-fusion, backend_config={"fusion_backend_config":{"kind":"__custom_fusion","custom_fusion_config":{"name":"dynamic_address_computation"}}}
+      one = s32[] constant(1)
+      i_plus_one = s32[] add(i, one)
+      loop_iter_plus_one = s32[] add(loop_iter, one)
+      ROOT tuple = tuple(i_plus_one, fusion, src, loop_iter)
+    }
+
+    Cond {
+      param = (s32[], s32[16,4], s32[8,4], s32[]) parameter(0)
+      loop_iter = s32[] get-tuple-element(param), index=0
+      four = s32[] constant(4)
+      ROOT compare = pred[] compare(loop_iter, four), direction=LT
+    }
+
+    ENTRY main {
+      zero = s32[] constant(0)
+      dest = s32[16,4] parameter(0)
+      src = s32[8,4] parameter(1)
+      tuple = tuple(zero, dest, src, zero)
+      ROOT while = (s32[], s32[16,4], s32[8,4], s32[]) while(tuple), body=Body, condition=Cond
+    }
+  )";
+
+  const char* normal = R"(
+    HloModule test_module, replica_count=2
+
+    add {
+      a = s32[] parameter(0)
+      b = s32[] parameter(1)
+      ROOT add = s32[] add(a, b)
+    }
+
+    Body {
+      param = (s32[], s32[16, 4], s32[8, 4], s32[]) parameter(0)
+      i = s32[] get-tuple-element(param), index=0
+      dest = s32[16,4] get-tuple-element(param), index=1
+      src = s32[8,4] get-tuple-element(param), index=2
+      loop_iter = s32[] get-tuple-element(param), index=3
+      offset_values = u64[4] constant({0,4,8,18446744073709551615})
+      offset_as_array = u64[1] dynamic-slice(offset_values, i), dynamic_slice_sizes={1}
+      offset = u64[] reshape(offset_as_array)
+      zero = u64[] constant(0)
+      rs = s32[4,4] reduce-scatter(src), channel_id=0, replica_groups={{0,1}}, use_global_device_ids=true, dimensions={0}, to_apply=add
+      fusion = s32[16,4] dynamic-update-slice(dest, rs, offset, zero)
+      one = s32[] constant(1)
+      i_plus_one = s32[] add(i, one)
+      loop_iter_plus_one = s32[] add(loop_iter, one)
+      ROOT tuple = tuple(i_plus_one, fusion, src, loop_iter)
+    }
+
+    Cond {
+      param = (s32[], s32[16,4], s32[8,4], s32[]) parameter(0)
+      loop_iter = s32[] get-tuple-element(param), index=0
+      four = s32[] constant(4)
+      ROOT compare = pred[] compare(loop_iter, four), direction=LT
+    }
+
+    ENTRY main {
+      zero = s32[] constant(0)
+      dest = s32[16,4] parameter(0)
+      src = s32[8,4] parameter(1)
+      tuple = tuple(zero, dest, src, zero)
+      ROOT while = (s32[], s32[16,4], s32[8,4], s32[]) while(tuple), body=Body, condition=Cond
+    }
+  )";
+
+  ErrorSpec error_spec{/*aabs=*/1e-3, /*arel=*/1e-3};
+  EXPECT_DEATH(
```

Comment:
This fails internally with

```
ignoring return value of function declared with 'nodiscard' attribute [-Werror,-Wunused-result]
 3406 |       RunAndCompareTwoModulesReplicated(hlo, normal, true, true, error_spec),
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./third_party/googletest/googletest/include/gtest/gtest-death-test.h:194:15: note: expanded from macro 'EXPECT_DEATH'
  194 |   EXPECT_EXIT(statement, ::testing::internal::ExitedUnsuccessfully, matcher)
```

This is the first time I see it, not even sure what's going on

Reply:
Let me see if I can fix this.